### ISSUE TITLE
move "ClangBuild" to code checks? 

### ISSUE THREAD

**Issue by VinInn** [cms-sw/cmssw#37950]
I suggest to move "ClangBuild" to code checks pass.

**Comment by makortel** [comment]
@VinInn Could you explain what gains are you looking for with that?

**Comment by VinInn** [comment]
It's a check of code syntax. Avoids one starts the tests and then discovers that has to modify something "trivial" in the code.

**Comment by VinInn** [comment]
Yes I submitted this issue after getting the ```-Wpessimizing-move```.
Sorry for not having quote prevously.

**Comment by smuzaffar** [comment]
`clang-format` does not catch any syntax issue but for `clang-tidy` code should be clean (buildable) so it should catch any syntax errors. `code-checks` part of PR tests just makes sure that the code you have touched has the proper format and passes clang-tidy checks. For this purpose there is no need to checkout the dependenct packages.
